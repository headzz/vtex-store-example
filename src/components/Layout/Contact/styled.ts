import { PINK_JUSSI } from 'configuration/constants'
import styled from 'styled-components'

export const ContactStyled = styled.section`
  background-color: ${PINK_JUSSI};
  padding: 13.6rem 0;
  width: 100%;
`
