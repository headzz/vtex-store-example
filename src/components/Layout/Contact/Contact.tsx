import { ContactStyled } from './styled'

/**
 * @component
 * @name Contact
 * @description Defines the container for the layout of contact section.
 * @namespace components components/Layout/Contact
 * @param {element} children Component(s) that may appear inside the layout.
 *
 */

const Contact: React.FC = ({ children }) => {
  return <ContactStyled>{children}</ContactStyled>
}

export default Contact
