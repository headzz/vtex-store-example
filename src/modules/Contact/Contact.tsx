import Layout from 'components/Layout'

import TextArea from './components/TextArea'
import { ContactContainer } from './styled'

/**
 * @component
 * @name Contact
 * @description creates Contact component section to use on screen.
 * @namespace components modules/Contact
 *
 */
const Contact: React.FC = () => (
  <Layout.Contact>
    <ContactContainer>
      <TextArea />
    </ContactContainer>
  </Layout.Contact>
)

export default Contact
