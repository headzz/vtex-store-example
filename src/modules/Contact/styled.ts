import styled from 'styled-components'

export const ContactContainer = styled.div`
  margin: 0 auto;
  div {
    padding: 0 1rem;
    width: 100%;

    p:first-child {
      margin-bottom: 3.2rem;
      text-align: center;
    }

    p:nth-child(2) {
      margin-bottom: 0.8rem;
    }
  }

  @media (min-width: 768px) {
    div {
      padding: 0;
      width: 67.6rem;
    }
  }
`
