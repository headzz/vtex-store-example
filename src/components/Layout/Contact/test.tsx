import { render } from '@testing-library/react'

import Contact from './Contact'

describe('<Contact />', () => {
  it('should render component that wraps Contact on page', () => {
    const { container } = render(<Contact />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
