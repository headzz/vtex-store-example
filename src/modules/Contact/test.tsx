import { render } from '@testing-library/react'

import Contact from './Contact'

describe('Contact', () => {
  it('match snapshot', () => {
    const { container } = render(<Contact />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
