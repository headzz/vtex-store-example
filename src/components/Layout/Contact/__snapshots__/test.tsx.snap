// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> should render component that wraps Contact on page 1`] = `
.c0 {
  background-color: #EE0E7D;
  padding: 13.6rem 0;
  width: 100%;
}

<section
  class="c0"
/>
`
